crates/sui-graphql-rpc/src/types/chain_identifier.rs

```diff
@@ -4,33 +4,41 @@
 use crate::{
     data::{Db, DbConnection, QueryExecutor},
     error::Error,
+    server::watermark_task::ChainIdentifierLock,
 };
 use async_graphql::*;
-use diesel::{ExpressionMethods, QueryDsl};
+use diesel::{ExpressionMethods, OptionalExtension, QueryDsl};
 use sui_indexer::schema::checkpoints;
 use sui_types::{
     digests::ChainIdentifier as NativeChainIdentifier, messages_checkpoint::CheckpointDigest,
 };
 
-pub(crate) struct ChainIdentifier;
+#[derive(Clone, Copy, Debug, Default)]
+pub(crate) struct ChainIdentifier(pub(crate) NativeChainIdentifier);
 
 impl ChainIdentifier {
     /// Query the Chain Identifier from the DB.
-    pub(crate) async fn query(db: &Db) -> Result<NativeChainIdentifier, Error> {
+    pub(crate) async fn query(db: &Db) -> Result<Option<NativeChainIdentifier>, Error> {
         use checkpoints::dsl;
 
-        let digest_bytes = db
+        let Some(digest_bytes) = db
             .execute(move |conn| {
                 conn.first(move || {
                     dsl::checkpoints
```

Comment:
When do we plan to merge this into `main`? I'm working on a PR that indexes chain identifier into its own table so that would require changes on/about these three lines or so. But everywhere else should be no conflict.

Reply:
The goal is to land this together with @amnn's work on packages. As long as the backfill is ready, we could land it.
Happy to adjust the code base myself, or you could do on top?

To reverse the question, would that change go in together with the full new schema changes?